### PR TITLE
Avoid findUnusedPor() when picking up a port for JMX agent

### PR DESCRIPTION
**Pull request by findepi**
Fixes https://github.com/prestosql/presto/issues/2386
Supersedes https://github.com/airlift/airlift/pull/801

### CODE REVIEW

**Comment by electrum** [comment]
We could change the behavior to require specifying a port for JMX. With modern systems like k8s, port allocation is not a problem.

File: jmx/src/main/java/io/airlift/jmx/JmxAgent9.java [hunk 1]
```diff
@@ -80,40 +83,46 @@ else if (configuredRegistryPort != null) {
...
             startJmxAgent(registryPort, serverPort);
-            log.info("JMX agent started and listening on %s", address);
+            HostAndPort address = getAddress(registryPort);
+            log.info("JMX agent started and listening on %s", getAddress(registryPort));
+            url = toJmxUrl(address);
         }
         else {
-            log.info("JMX agent already running and listening on %s", address);
+            url = startJmxAgent(0, serverPort);
+            log.info("JMX agent started locally");
```
[the remark this answers is not in the document]
Comment (martint) [reply jmx/src/main/java/io/airlift/jmx/JmxAgent9.java:100]: Some experiments...

With this program:

```java
import java.io.IOException;
import java.net.InetSocketAddress;
import java.net.ServerSocket;

class Port
{
    public static void main(String[] args)
            throws IOException
    {
        while (true) {
            int port = findPort();
            try (ServerSocket socket = new ServerSocket()) {
                try {
                    socket.bind(new InetSocketAddress(port));
                    System.out.println("bound to: " + port);
                }
                catch (IOException e) {
                    System.out.println("port already in use: " + port);
                }
            }
        }
    }

    private static int findPort()
            throws IOException
    {
        try (ServerSocket socket = new ServerSocket(0)) {
            return socket.getLocalPort();
        }
    }
}
```

On Mac OS, ports are allocated sequentially. It generally works, but I occasionally see some failures to bind a port:

```
bound to: 58694
bound to: 58695
bound to: 58696
bound to: 58697
bound to: 58698
bound to: 58699
bound to: 58700
bound to: 58701
bound to: 58702
bound to: 58703
bound to: 58704
bound to: 58705
bound to: 58706
bound to: 58707
bound to: 58708
bound to: 58709
port already in use: 58710
bound to: 58711
bound to: 58712
```

On Linux, the ports are not allocated sequentially, but I haven't been able to see a single failure after thousands of runs.

```
bound to: 39093
bound to: 45791
bound to: 35941
bound to: 44555
bound to: 42219
bound to: 39907
bound to: 38633
bound to: 36583
bound to: 36131
bound to: 42735
bound to: 45505
bound to: 45791
bound to: 34109
bound to: 43093
bound to: 38935
bound to: 39353
bound to: 34079
bound to: 45291
bound to: 38801
bound to: 42465
bound to: 35429
```

**Comment by findepi** [comment]
> This is a change in behavior. If a port is not specified. 

Correct. My assumption is that after the change we still retain introspection.

what's the use-case of exposing agent on an arbitrary port?
no-one externally is going to use it anyway, right?

let's talk.

**Comment by findepi** [comment]
Discussed offline changes with @martint and adjusted accordingly.

**Comment by findepi** [comment]
@martint added a note

**Comment by martint** [comment]
Ship it!

**Comment by findepi** [comment]
Added CHANGES.

**Pull request merged**